python/ray/serve/_private/replica.py

```diff
@@ -243,9 +243,16 @@ async def reconfigure(
             if user_config is not None:
                 await self.replica.reconfigure(user_config)
 
-            return self.get_metadata()
+            return await self.get_metadata()
 
-        def get_metadata(self) -> Tuple[DeploymentConfig, DeploymentVersion]:
+        async def get_metadata(
+            self,
+        ) -> Tuple[DeploymentConfig, DeploymentVersion]:
+            # When the self.replica variable is None, potentially replica is still
+            # under initialization. Relying on DeploymentStateManager to
+            # timeout the replica.
```

Comment:
I'm a little lost at why this change is needed. Is it because the constructor might still be running while we are recovering?

Reply:
If so, can we use an explicit `asyncio.Event` instead of implementing our own sleeping logic here?

Reply:
e.g., `self._initialized = asyncio.Event()`

Reply:
> I'm a little lost at why this change is needed. Is it because the constructor might still be running while we are recovering?
yes

super, I forget this Async variable! updated.